creation/create_cvmfsexec_distros.sh

```diff
@@ -17,86 +17,103 @@ CVMFS_SOURCES=osg:egi:default
+        echo "Using (existing) cvmfsexec version `cat $cvmfsexec_tarballs/.cvmfsexec_version`"
+    fi
+    echo "Skipping the building of cvmfsexec distribution tarballs..."
+    rm -rf $cvmfsexec_latest
+    exit 0
+else
+    # if current version and latest version are different
+    if [[ -z "$curr_ver" ]]; then
+        # $curr_ver is empty; first time run of factory upgrade
+        # no version info stored in work-dir/cvmfsexec/tarballs
+        echo "Building cvmfsexec distributions..."
+    else
+        # $curr_ver is not empty; subsequent run of factory upgrade (and not the first time)
+        echo "Found newer version of cvmfsexec..."
+        echo "Rebuilding cvmfsexec distributions using the latest version..."
+    fi
 
-if [[ ! -d $cvmfsexec_tarballs ]]; then
-    mkdir -p $cvmfsexec_tarballs
-fi
+    # build the distributions for cvmfsexec based on the source, os and platform combination
+    cvmfsexec_distros=$cvmfsexec_temp/distros
+    if [[ ! -d $cvmfsexec_distros ]]; then
+        mkdir -p $cvmfsexec_distros
+    fi
 
-declare -a cvmfs_sources
-cvmfs_sources=($(echo $CVMFS_SOURCES | tr ":" "\n"))
+    if [[ ! -d $cvmfsexec_tarballs ]]; then
+        mkdir -p $cvmfsexec_tarballs
+    fi
 
-declare -a machine_types
-machine_types=($(echo $SUPPORTED_TYPES | tr ":" "\n"))
+    declare -a cvmfs_sources
+    cvmfs_sources=($(echo $CVMFS_SOURCES | tr ":" "\n"))
 
-for cvmfs_src in "${cvmfs_sources[@]}"
-do
-    for mach_type in "${machine_types[@]}"
+    declare -a machine_types
+    machine_types=($(echo $SUPPORTED_TYPES | tr ":" "\n"))
+
+    for cvmfs_src in "${cvmfs_sources[@]}"
     do
-        echo -n "Making $cvmfs_src distribution for $mach_type machine..."
-        os=`echo $mach_type | awk -F'-' '{print $1}'`
-        arch=`echo $mach_type | awk -F'-' '{print $2}'`
-        $cvmfsexec_base/makedist -m $mach_type $cvmfs_src &> /dev/null
-        if [[ $? -eq 0 ]]; then
-           $cvmfsexec_base/makedist -o $cvmfsexec_distros/cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-           if [[ -e $cvmfsexec_distros/cvmfsexec-${cvmfs_src}-${os}-${arch} ]]; then
-               echo " Success"
-               tar -cvzf $cvmfsexec_tarballs/cvmfsexec_${cvmfs_src}_${os}_${arch}.tar.gz -C $cvmfsexec_distros cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-           fi
-        else
-            echo " Failed! REASON: $cvmfs_src may not yet have a $mach_type build."
-        fi
-
-        # delete the dist directory within cvmfsexec to download the cvmfs configuration
-        # and repositories for another machine type
-        rm -rf $cvmfsexec_base/dist
+        for mach_type in "${machine_types[@]}"
+        do
+            echo -n "Making $cvmfs_src distribution for $mach_type machine..."
+            os=`echo $mach_type | awk -F'-' '{print $1}'`
+            arch=`echo $mach_type | awk -F'-' '{print $2}'`
+            $cvmfsexec_latest/makedist -m $mach_type $cvmfs_src &> /dev/null
+            if [[ $? -eq 0 ]]; then
```

Comment:
Better in one line and remember to quote the paths: `if "$cvmfsexec_latest"/makedist -m $mach_type $cvmfs_src &> /dev/null ; then`

Reply:
updated